tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_consolidation_request.py

```diff
@@ -887,6 +887,38 @@ def test_incorrect_target_with_bls_credential(spec, state):
     )
 
 
+@with_electra_and_later
+@with_presets([MINIMAL], "need sufficient consolidation churn limit")
+@with_custom_state(
+    balances_fn=scaled_churn_balances_exceed_activation_exit_churn_limit,
+    threshold_fn=default_activation_threshold,
+)
+@spec_test
+@single_phase
+def test_incorrect_source_with_bls_credential(spec, state):
+    # move state forward SHARD_COMMITTEE_PERIOD epochs to allow for consolidation
+    state.slot += spec.config.SHARD_COMMITTEE_PERIOD * spec.SLOTS_PER_EPOCH
+
+    current_epoch = spec.get_current_epoch(state)
+    source_index = spec.get_active_validator_indices(state, current_epoch)[0]
+    target_index = spec.get_active_validator_indices(state, current_epoch)[1]
+    set_compounding_withdrawal_credential_with_balance(spec, state, target_index)
+
+    # It's not possible to control this field, but if you were somehow able to control
+    # the address which shared the same bytes as the validator's BLS pubkey.
```

Comment:
```suggestion
```

We verified with @nflaig and @jtraglia that this is in fact controllable by a potential attacker. There is no validation on the withdrawal credentials, and someone could effectively make a deposit with a BLS prefix while their withdrawal credentials end with an address that they can control in order to make a consolidation request from this same address.

Reply:
Indeed, thanks for submitting this finding @iFrostizz 🙂

Reply:
And I've updated this comment so this is clear. How do you feel about the new comment?